drgn/helpers/linux/slab.py

```diff
@@ -53,3 +55,163 @@ def print_slab_caches(prog: Program) -> None:
     for s in for_each_slab_cache(prog):
         name = escape_ascii_string(s.name.string_(), escape_backslash=True)
         print(f"{name} ({s.type_.type_name()})0x{s.value_():x}")
+
+
+def obtain_freelist_slub(page_or_slab: Object, offset: int) -> Set[Object]:
+    """slub freelist is a pointer to the first free object"""
+    freelist_set = set()
+    freelist = page_or_slab.freelist.read_()
+
+    while freelist:
+        freelist_set.add(freelist.value_())
+        freelist = cast("void **", freelist + offset)[0].read_()
+    return freelist_set
+
+
+def obtain_freelist_slab(
+    prog: Program, page_or_slab: Object, freelist_index_size: int
+) -> Set[int]:
+    """slab freelist an array of free object indices"""
+    freelist_set = set()
+    freelist = page_or_slab.freelist.read_()
```

Comment:
Rather than reading the raw bytes of the `freelist`, can you cast the `freelist` to `freelist_idx_t` and use it just like an array?

(You might want to make this a local function so that you only need to look up `freelist_idx_t` and don't need to pass it around, too).